### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by alexsu52**
Updated Readme.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -78,145 +135,201 @@ See [third_party_integration](./third_party_integration) for examples of code mo
...
+  - TensorFlow\* 2.4.2
 
-*NOTE:* The best known PyTorch version for the current NNCF is 1.8.1, and it is highly recommended to use it.
-
-^ If a torch package built for specific CUDA version is already present in the environment into which NNCF is being installed,
-and if it has a matching base version, then the CUDA version for which the present torch package is targeted will be used.
-
-Otherwise NNCF will install the latest available torch version from pip, which is targeted to the CUDA version of PyTorch packaging strategy's choosing.
-For PyTorch 1.8.1, the default CUDA is 10.2.
+This repository is tested on Python* 3.6.2+, PyTorch* 1.8.1 (NVidia CUDA\* Toolkit 10.2) and TensorFlow* 2.4.2 (NVidia CUDA\* Toolkit 11.0).
```
[the remark this answers is not in the document]
Comment (vshampor) [reply README.md:142]: You are mixing the driver-reported capabilities with the CUDA runtime lib version that is actually used. What actually gets used is the 10.2, this is the default for 1.8.1

File: README.md [hunk 2]
```diff
@@ -78,145 +135,201 @@ See [third_party_integration](./third_party_integration) for examples of code mo
...
-|ResNet-18|XNOR (weights), scale/threshold (activations)|ImageNet|69.8|61.63|
-|ResNet-18|DoReFa (weights), scale/threshold (activations)|ImageNet|69.8|61.61|
-|ResNet-18|Filter pruning, 40%, magnitude criterion|ImageNet|69.8|69.26|
-|ResNet-18|Filter pruning, 40%, geometric median criterion|ImageNet|69.8|69.32|
-|ResNet-34|Filter pruning, 40%, geometric median criterion|ImageNet|73.3|72.73|
-|GoogLeNet|Filter pruning, 40%, geometric median criterion|ImageNet|69.75|68.82|
-
+|PyTorch Model|<img width="115" height="1">Compression algorithm<img width="115" height="1">|Dataset|Accuracy (Drop) %|
+| :---: | :---: | :---: | :---: |
+|ResNet-50|INT8|ImageNet|76.42 (-0.26)|
```
Comment (vshampor) [inline README.md:215]: Perhaps we should specify the drop in a separate column? Anyway, could you please update the `tools/update_eval_results.py` so that it generates tables consistent with any of the two new formats?

File: README.md [hunk 3]
```diff
@@ -1,38 +1,56 @@
 # Neural Network Compression Framework (NNCF)
 
-This repository contains a PyTorch\*-based framework and samples for neural networks compression.
+NNCF provides a suite of advanced compression algorithms, applied during a model fine-tuning and samples for models 
+compression. It is highly aligned with [OpenVINO&trade;](https://github.com/openvinotoolkit/openvino) and is aimed 
+to improve runtime performance of models in OpenVINO&trade; with minimal accuracy drop.
 
-The framework is organized as a Python\* package that can be built and used in a standalone mode.
-The framework architecture is unified to make it easy to add different compression methods.
+NNCF is designed to work with models from the two most popular deep learning frameworks, [PyTorch](https://pytorch.org/)
```
Comment (MaximProshin) [inline README.md:7]: I would remove "the two most popular deep learning frameworks," and simply say "from PyTorch and TensorFlow"

File: README.md [hunk 4]
```diff
@@ -1,38 +1,56 @@
...
-- Common interface for compression methods
-- GPU-accelerated layers for faster compressed model fine-tuning
-- Distributed training support
+- Support of various compression algorithms, applied during a model fine-tuning process to achieve best compression 
+  parameters and accuracy:
+  
+  |Compression algorithm|PyTorch|TensorFlow|
+  | :--- | :---: | :---: |
+  |[Quantization](./docs/compression_algorithms/Quantization.md) | Supported | Supported |
+  |[Mixed-Precision Quantization](./docs/compression_algorithms/Quantization.md#mixed_precision_quantization) | Supported | Support planned |
```
Comment (MaximProshin) [inline README.md:27]: I think it's fair to state "Not supported" for Mixed-Precision and mark it as Supported when implemented.

File: README.md [hunk 5]
```diff
@@ -78,145 +135,201 @@ See [third_party_integration](./third_party_integration) for examples of code mo
 ## System requirements
 - Ubuntu\* 18.04 or later (64-bit)
 - Python\* 3.6.2 or later
-- NVidia CUDA\* Toolkit 10.2 or later^
-- PyTorch\* 1.5 or later (1.8.0 not supported, 1.8.1 supported)
+- Supported frameworks:
```
Comment (MaximProshin) [inline README.md:138]: I think it should be rephrased as there is no support for versions >1.8.1

File: README.md [hunk 6]
```diff
@@ -78,145 +135,201 @@ See [third_party_integration](./third_party_integration) for examples of code mo
...
-
-^ If a torch package built for specific CUDA version is already present in the environment into which NNCF is being installed,
-and if it has a matching base version, then the CUDA version for which the present torch package is targeted will be used.
-
-Otherwise NNCF will install the latest available torch version from pip, which is targeted to the CUDA version of PyTorch packaging strategy's choosing.
-For PyTorch 1.8.1, the default CUDA is 10.2.
+This repository is tested on Python* 3.6.2+, PyTorch* 1.8.1 (NVidia CUDA\* Toolkit 10.2) and TensorFlow* 2.4.2 (NVidia CUDA\* Toolkit 11.0).
 
 ## Installation
 We suggest to install or use the package in the [Python virtual environment](https://docs.python.org/3/tutorial/venv.html).
```
Comment (MaximProshin) [inline README.md:145]: These two sentences about virtual environment could be combined.

File: README.md [hunk 7]
```diff
@@ -78,145 +135,201 @@ See [third_party_integration](./third_party_integration) for examples of code mo
...
 
+First, create a virtual environment with the version of Python you're going to use and activate it.
 
-#### As a package built from a checked-out repository:
-1) Install the following system dependencies:
-
-`sudo apt-get install python3-dev`
+Then you need to install one of the deep learning frameworks [PyTorch](https://pytorch.org/) or [TensorFlow](https://www.tensorflow.org/). Please follow to [TensorFlow installation guide](https://www.tensorflow.org/install/) or [PyTorch installation guide](https://pytorch.org/get-started/locally/#start-locally).
```
Comment (MaximProshin) [inline README.md:149]: I propose to rephrase it a little like "If you want to optimize a model from PyTorch, install ..."

File: README.md [hunk 8]
```diff
@@ -78,145 +135,201 @@ See [third_party_integration](./third_party_integration) for examples of code mo
...
 
-2) Install the package and its dependencies by running the following in the repository root directory:
+When one of those backends has been installed, NNCF can be installed as follow:
```
Comment (MaximProshin) [inline README.md:151]: I think it can be removed

File: README.md [hunk 9]
```diff
@@ -78,145 +135,201 @@ See [third_party_integration](./third_party_integration) for examples of code mo
...
+When one of those backends has been installed, NNCF can be installed as follow: 
 
-`python setup.py install`
+#### As a package built from a checked-out repository:
 
+Install the package and its dependencies by running the following in the repository root directory:
+```
+python setup.py install
+```
+Alternatively, If you don't install any backend you can install NNCF and PyTorch in one line with:
```
Comment (MaximProshin) [inline README.md:159]: This seems to be strange considering that above it's stated "you need to install one of the deep learning frameworks"

**Review by MaximProshin — CHANGES_REQUESTED** [review]

File: README.md [hunk 10]
```diff
@@ -78,145 +135,201 @@ See [third_party_integration](./third_party_integration) for examples of code mo
...
-and if it has a matching base version, then the CUDA version for which the present torch package is targeted will be used.
-
-Otherwise NNCF will install the latest available torch version from pip, which is targeted to the CUDA version of PyTorch packaging strategy's choosing.
-For PyTorch 1.8.1, the default CUDA is 10.2.
+This repository is tested on Python* 3.6.2+, PyTorch* 1.8.1 (NVidia CUDA\* Toolkit 10.2) and TensorFlow* 2.4.2 (NVidia CUDA\* Toolkit 11.0).
 
 ## Installation
 We suggest to install or use the package in the [Python virtual environment](https://docs.python.org/3/tutorial/venv.html).
 
+First, create a virtual environment with the version of Python you're going to use and activate it.
```
Comment (MaximProshin) [inline README.md:147]: These two sentences about virtual environment could be combined.

File: README.md [hunk 11]
```diff
@@ -1,38 +1,56 @@
 # Neural Network Compression Framework (NNCF)
 
-This repository contains a PyTorch\*-based framework and samples for neural networks compression.
+NNCF provides a suite of advanced compression algorithms, applied during a model fine-tuning and samples for models
```
Comment (AlexKoff88) [inline README.md:3]: I would add something about optimization for inference. Sometimes people get confused from the beginning. Like, `NNCF provides a suite of advanced algorithms for Neural Networks inference optimization...`.

File: README.md [hunk 12]
```diff
@@ -1,38 +1,56 @@
...
-    - [Quantization](./docs/compression_algorithms/Quantization.md)
-    - [Binarization](./docs/compression_algorithms/Binarization.md)
-    - [Sparsity](./docs/compression_algorithms/Sparsity.md)
-    - [Filter pruning](./docs/compression_algorithms/Pruning.md)
-- Automatic, configurable model graph transformation to obtain the compressed model. The source model is wrapped by the custom class and additional compression-specific layers are inserted in the graph.
-- Common interface for compression methods
-- GPU-accelerated layers for faster compressed model fine-tuning
-- Distributed training support
+- Support of various compression algorithms, applied during a model fine-tuning process to achieve best compression 
+  parameters and accuracy:
```
Comment (AlexKoff88) [inline README.md:22]: ` ...to achieve best compression parameters and accuracy` is not fully clear and justified to me. Let's remove or rephrase, like, `...applied during a model fine-tuning process to achieve a better performance-accuracy trade-off.`

File: README.md [hunk 13]
```diff
@@ -1,38 +1,56 @@
...
+  |[Sparsity](./docs/compression_algorithms/Sparsity.md) | Supported | Supported |
+  |[Filter pruning](./docs/compression_algorithms/Pruning.md) | Supported | Supported |
+
+- Automatic, configurable model graph transformation to obtain the compressed model.
+  > **NOTE**: Limited support for TensorFlow models. The models created using Sequential or Keras Functional API are only supported.
+- Common interface for compression methods.
+- GPU-accelerated layers for faster compressed model fine-tuning.
+- Distributed training support.
 - Configuration file examples for each supported compression algorithm.
 - Git patches for prominent third-party repositories ([mmdetection](https://github.com/open-mmlab/mmdetection), [huggingface-transformers](https://github.com/huggingface/transformers)) demonstrating the process of integrating NNCF into custom training pipelines
```
Comment (AlexKoff88) [inline README.md:38]: Is mmdetection patch still available? I believe the collective decision was to remove it and refer for an OTE fork of mmdetection.

File: README.md [hunk 14]
```diff
@@ -42,33 +60,72 @@ model = resnet50()
...
 
 For a quicker start with NNCF-powered compression, you can also try the sample scripts, each of which provides a basic training pipeline for classification, semantic segmentation and object detection neural network training correspondingly.
 
 To run the samples please refer to the corresponding tutorials:
-- [Image Classification sample](examples/torch/classification/README.md)
-- [Object Detection sample](examples/torch/object_detection/README.md)
-- [Semantic Segmentation sample](examples/torch/semantic_segmentation/README.md)
+
+- PyTorch samples:
+  - [Image Classification sample](examples/torch/classification/README.md)
```
Comment (AlexKoff88) [inline README.md:122]: Link to OTE and short description are required!

File: README.md [hunk 15]
```diff
@@ -78,145 +135,201 @@ See [third_party_integration](./third_party_integration) for examples of code mo
 ## System requirements
 - Ubuntu\* 18.04 or later (64-bit)
 - Python\* 3.6.2 or later
-- NVidia CUDA\* Toolkit 10.2 or later^
-- PyTorch\* 1.5 or later (1.8.0 not supported, 1.8.1 supported)
+- Supported frameworks:
+  - PyTorch\* 1.5 or later (1.8.0 not supported, 1.8.1 supported)
```
[the remark this answers is not in the document]
Comment (alexsu52) [reply README.md:139]: @ljaljushkin Thanks for your comment. There is torch>=1.5.0, <=1.8.1, !=1.8.0" condition in the `setup.py`. @vshampor, what version Pytorch we should specify in the Readme? `PyTorch\* >=1.5.0, <=1.8.1 (1.8.0 not supported)` or `PyTorch\* 1.8.1`?

File: README.md (hunk 2 above)
Comment (alexsu52) [reply README.md:215]: @AlexKoff88, will we use `Accuracy (Drop) %` format for the main Readme?

File: README.md (hunk 5 above)
Comment (alexsu52) [reply README.md:138]: What do you think about the following formulation `PyTorch\* >=1.5.0, <=1.8.1 (1.8.0 not supported)`?

File: README.md (hunk 9 above)
Comment (alexsu52) [reply README.md:159]: I believe we should point that we have extra dependency options for nncf package and the user can install nncf + torch using nncf[torch] and nncf + tensorflow using nncf[tf]. If you have a better formulation or you have suggestions to reorganize install topic, let me know.

File: README.md (hunk 13 above)
Comment (alexsu52) [reply README.md:38]: @vshampor will we delete mmdetection here or will you make a separate PR?

File: README.md (hunk 3 above)
Comment (alexsu52) [reply README.md]: Done

File: README.md (hunk 4 above)
Comment (alexsu52) [reply README.md]: Done

File: README.md (hunk 7 above)
Comment (alexsu52) [reply README.md]: Done

File: README.md (hunk 8 above)
Comment (alexsu52) [reply README.md]: Done

File: README.md (hunk 10 above)
Comment (alexsu52) [reply README.md]: Done

File: README.md (hunk 11 above)
Comment (alexsu52) [reply README.md]: I have reformulated it, please take a look.

File: README.md (hunk 12 above)
Comment (alexsu52) [reply README.md]: Done

File: README.md (hunk 13 above)
Comment (vshampor) [reply README.md:34]: You may delete it here, @aleksei-kashapov was supposed to make a PR, but did not have the time for it, it seems.

File: README.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (vshampor) [reply README.md:135]: 1.8.1 is BKC, but we "should" support the others in setup.py as wel, so I think the former variant should be specified.

File: README.md (hunk 5 above)
Comment (MaximProshin) [reply README.md:134]: It's ok with me

File: README.md (hunk 9 above)
Comment (MaximProshin) [reply README.md:152]: I understand why you added it but it looks a bit strange. One option to improve it is to state in the beginning: 
There are two ways how you could install NNCF:
1. Install pt or tf framework and then nncf
2. Install nncf[pt] which will also install pt framework or nncf[tf]...

**Review by MaximProshin — APPROVED** [review]

File: README.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (ljaljushkin) [reply README.md:135]: IMHO, user will be confused with the former variant: `PyTorch\* 1.5 or later (1.8.0 not supported, 1.8.1 supported)`

PyTorch 1.5 or **later** is supported, but 1.8.1 is somehow **more supported**? What's the difference?
Does it mean that 1.7 and 1.9 equally supported?

1.9 is not working for sure. We shouldn't recommend to install it. 
And we can't be 100% sure that 1.7 is also supported, because we don't test it. 

I wouldn't mislead users with unconfirmed claims. 
We can explicitly say that we tested NNCF for PT with 1.8.1. You could try other versions, but we don't guarantee operability.

[the remark this answers is not in the document]
Comment (alexsu52) [reply README.md:135]: @vshampor, @ljaljushkin Thanks for your comments. I have updated to `PyTorch\* >=1.5.0, <=1.8.1 (1.8.0 not supported)`. I hope the following note clears up any misunderstandings: 
```
 This repository is tested on Python* 3.6.2+, PyTorch* 1.8.1 (NVidia CUDA\* Toolkit 10.2) and TensorFlow* 2.4.2 (NVidia CUDA\* Toolkit 11.0).
```

File: README.md (hunk 9 above)
Comment (MaximProshin) [reply README.md:152]: Let's keep it as it is. My comment was about "install", now I see that the same is applicable for PyPI so my suggestion won't make it clearer.

File: README.md (hunk 13 above)
Comment (alexsu52) [reply README.md]: Done

File: README.md (hunk 14 above)
Comment (alexsu52) [reply README.md:118]: I have added, please take a look.

**Review by vshampor — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (AlexKoff88) [reply README.md:214]: I support, this is very informative.

**Review by AlexKoff88 — APPROVED** [review]

**Comment by alexsu52** [comment]
Jenkins please retry a build
